Automattic/wp-calypso
# My Sites: add Jetpack Search section

#### Changes proposed in this Pull Request

* Add 'Jetpack' -> 'Search' to sidebar
* Add a page at `/jetpack-search/<slug>` either suggesting that the user should sign up for JP Search, or showing their current subscription

Fixes #44315.

Without Jetpack Search subscription:

<img width="814" alt="Screen Shot 2020-11-25 at 12 08 29" src="https://user-images.githubusercontent.com/17325/100289565-149cd600-2fde-11eb-81be-73bc476517af.png">

With Jetpack Search subscription:

<img width="828" alt="Screen Shot 2020-11-25 at 15 16 55" src="https://user-images.githubusercontent.com/17325/100289585-1c5c7a80-2fde-11eb-98d0-72da5108a94d.png">

#### Testing instructions



Navigate to My Sites > Jetpack > Search. Try for a site with a Jetpack Search subscription, and for one without.

You can also try navigating to http://calypso.localhost:3000/jetpack-search and ensure that you're prompted to choose a site.
